File: src/main/java/com/mbayou/kick4k/users/TokenIntrospect.java
package com.mbayou.kick4k.users;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class TokenIntrospect {
    private final Boolean active;
    private final String clientId;
    private final Integer exp;
    private final String scope;
    private final String tokenType;

    @JsonCreator
    public TokenIntrospect(@JsonProperty("active") Boolean active,
                           @JsonProperty("client_id") String clientId,
                           @JsonProperty("exp") Integer exp,
                           @JsonProperty("scope") String scope,
                           @JsonProperty("token_type") String tokenType) {
        this.active = active;
        this.clientId = clientId;
        this.exp = exp;
        this.scope = scope;
        this.tokenType = tokenType;
    }

    public Boolean isActive() {
        return active;
    }

    public String getClientId() {
        return clientId;
    }

    public Integer getExp() {
        return exp;
    }

    public String getScope() {
        return scope;
    }

    public String getTokenType() {
        return tokenType;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/users/TokenIntrospect.kt
package com.mbayou.kick4k.users

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class TokenIntrospect @JsonCreator constructor(
    @JsonProperty("active") val active: Boolean?,
    @JsonProperty("client_id") val clientId: String?,
    @JsonProperty("exp") val exp: Int?,
    @JsonProperty("scope") val scope: String?,
    @JsonProperty("token_type") val tokenType: String?,
)
